docs/02-guides/customize-wysiwyg-platform.mdx

```diff
@@ -0,0 +1,200 @@
+theme.
+
+Now, within your newly created `getWidgetComponent.js` file, you will be able to
+add your own behavior. For instance, if you've created a widget in Magento that
+uses the type `acme/product-preview`, you will need to update the `Widget.js`
+file likewise:
+
+```js title="theme/modules/Wysiwyg/MagentoWysiwyg/Widget/getWidgetComponent.js"
+import React from "react";
+import Spinner from "theme/components/atoms/Spinner";
+import loadable from "@loadable/component";
+
+const defaultWidgetsMap = {
+  "acme/product-preview": loadable(
+    () => import("./path/to/ProductPreview.js"),
+    {
+      fallback: <Spinner />,
+    }
+  ),
+};
+
+// ... rest of the file is intact
+```
+
+If you restart your application, you will notice that the new widget component
+is displayed. Hurray!
+
+However, in most cases, you will need to fetch data to display all the needed
+information in your widget. For instance, if the widget is
+`{% raw %}{{ widget type="acme/product-preview" sku="VSK12" }}{% endraw %}` you
+will want to fetch the product associated with the given SKU.
+
+#### Register your widget type at the GraphQL level
+
+1.  First make sure that your dependencies are up to date in your GraphQL module
+    ```ts title="extensions/acme-extension/modules/wysiwyg/index.ts"
+    export default createGraphQLModule({
+    namespace: "Magento2/Cms",
+    // highlight-start
+    dependencies: [
+      "Magento2/Wysiwyg", // ensure that Widget related features are available
+      "Magento2/Catalog/Product" // ensure that you can fetch a product in your Wysiwyg data
+    ],
+    // highlight-end
+    ```
+1.  Then you need to register a new WidgetData type in your schema
+    ```ts title="extensions/acme-extension/modules/wysiwyg/index.ts"
+    typeDefs: /* GraphQL */ `
+      type WidgetProductPreviewData implements WidgetData {
+        dataId: ID
+        product: Product
+      }
+    `,
+    ```
+1.  After that you can setup the resolvers to tell GraphQL how to fetch the
+    `product` field
+    ```ts title="extensions/acme-extension/modules/wysiwyg/resolvers.ts"
+    export default {
+      WidgetProductPreviewData: {
+        product: ({ node }, _, { loaders }) => {
+          const productSkuAttribute = node.attrs.find(
+            ({ name }) => name === "sku"
+          );
+          if (!productSkuAttribute) {
+            return null;
+          }
+          return loaders.Product.load(productSkuAttribute.value);
+        },
+      },
+    };
+    ```
+1.  Finally register the widget type from Magento and associate it with the
+    GraphQL type
+
+    ```ts title="extensions/acme-extension/modules/wysiwyg/runtime.ts"
+      // ...
+      contextEnhancer: ({ loaders }) => {
+        loaders.MagentoWidget.registerWidgetType(
+          // the tag name in your HTML
+          "acme/product-preview",
+          // The associated GraphQL type name
+          "WidgetProductPreviewData"
+        );
+      },
+      // ...
```

Comment:
Show the full example;
- missing the `loadRuntime` in the GraphqlModule
- missing the import and creatpr function for `createGraphQLRuntime({...})`